index.d.ts

```diff
@@ -0,0 +1,76 @@
+// Type definitions for express-session 1.16
+// Project: https://github.com/expressjs/session
+// Definitions by: Hiroki Horiuchi <https://github.com/horiuchi>
+//                 Jacob Bogers <https://github.com/jacobbogers>
+//                 Naoto Yokoyama <https://github.com/builtinnya>
+//                 Ryan Cannon <https://github.com/ry7n>
+//                 HoldYourWaffle <https://github.com/HoldYourWaffle>
+
+import express from 'express'
+import node from 'events'
+
+
+
+declare global {
+  namespace Express {
+    interface Request { //inject additional properties on express.Request
+      session?: Session;
+      sessionID?: string;
+    }
+  }
+
+  export interface SessionData {
+    //seperate interface to allow application-specific declaration merging (see README)
+  }
+}
+
+
+
+interface SessionOptions {
+  secret: string | string[];
+
+  genid?(req: express.Request): string;
+  name?: string;
+  store?: Store;
+  cookie?: express.CookieOptions;
+  rolling?: boolean;
+  resave?: boolean;
+  proxy?: boolean;
+  saveUninitialized?: boolean;
+  unset?: string;
+}
+
+export default function session(options?: SessionOptions): express.RequestHandler;
+
+
+
+interface Session extends SessionData {
+  id: string;
+  cookie: SessionCookie;
+
+  regenerate(callback: (err: any) => void): Session;
+  destroy(callback: (err: any) => void): Session;
+  reload(callback: (err: any) => void): Session;
+  save(callback?: (err: any) => void): Session;
+  touch(): Session;
+}
+
+interface SessionCookie extends express.CookieOptions { //XXX actually a class
+  originalMaxAge: number;
+}
+
+
+
+export abstract class Store extends node.EventEmitter {
+  constructor(config?: any);
```

Comment:
Is this a reference to the `Store` class in this module? There are no arguments to the constructor if so: https://github.com/expressjs/session/blob/master/session/store.js#L32

Reply:
Actually all these methods are different, so not sure what this references and then where the declaration to the `Store` class is.

Reply:
> Is this a reference to the Store class in this module? There are no arguments to the constructor if so

It is. Seems like the DT definitions are wrong again (although I probably should've checked before copying it).

> Actually all these methods are different, so not sure what this references and then where the declaration to the Store class is.

It was meant to be an 'abstract base class' (like in Java) for Store implementations. Turns out I understood abstract classes in TypeScript wrong. I do know now where `regenerate`, `load` and `createSession` are coming from, I must've completely overlooked them at first.

I have updated the definitions with the proper abstract class definition.